nrf_modem/doc/ug_nrf_modem_porting_os.rst

```diff
@@ -53,27 +53,31 @@ nrf_modem_os_timedwait
 This function is called by the Modem library when a timed context or sleep is required.
 A blind return value of 0 will make all the Modem library operations always blocking.
 
-Note: Calls to :c:func:`nrf_modem_os_timedwait` must be able to enter a waiting state regardless of the modem library and modem OS initialization status. Hence, it must be possible to call :c:func:`nrf_modem_os_timedwait` before any call to :c:func:`nrf_modem_os_init`. To ensure that all waiting threads are woken on :c:func:`nrf_modem_event_notify`, the waiting threads must not be cleared on :c:func:`nrf_modem_os_init` or :c:func:`nrf_modem_os_shutdown`.
+.. note::
+   The calls to :c:func:`nrf_modem_os_timedwait` function must be able to enter a waiting state regardless of the modem library and modem OS initialization status.
+   Hence, it must be possible to call the :c:func:`nrf_modem_os_timedwait` function before any call to the :c:func:`nrf_modem_os_init` function.
+   To ensure all waiting threads are awake on :c:func:`nrf_modem_event_notify` function, the waiting threads must not be cleared on :c:func:`nrf_modem_os_init` or :c:func:`nrf_modem_os_shutdown` functions.
 
-*Required actions* to make the operations non-blocking:
+*Required actions* to be taken that do not block the operations:
 
 * Start counting the time (this can be based on a Timer or Thread for instance).
-* In the case of a call to :c:func:`nrf_modem_os_event_notify` all waiting threads should wake up, reporting back the remaining time of the timer through the :c:var:`timeout` variable.
+* In the case of a call to :c:func:`nrf_modem_os_event_notify` function, all waiting threads must wake up, reporting back the remaining time of the timer through the :c:var:`timeout` variable.
+
+The following points decides the *Function return value*:
 
-*Function return value* is decided by the following:
-* If the modem is not initialized, that is if :c:func:`nrf_modem_is_initialized` return false, return -NRF_ESHUTDOWN.
-* If timed out, return -NRF_ETIMEDOUT.
-* Else return 0.
+* If the modem is not initialized, that is if :c:func:`nrf_modem_is_initialized` return false, return ``-NRF_ESHUTDOWN``.
+* If there is a time out, return ``-NRF_ETIMEDOUT``.
+* Else return ``0``.
 
 nrf_modem_os_event_notify
 =========================
 
-This function is called by the Modem library when an Event occour and all threads waiting in :c:func:`nrf_modem_os_timedwait` should wake up.
+This function is called by the Modem library when an event occurs and all threads waiting for :c:func:`nrf_modem_os_timedwait` function wakes up.
```

Comment:
This is not correct anymore. The threads do not wait _for_ `nrf_modem_os_timedwait()`, they wait _in_ `nrf_modem_os_timedwait()`, as the original text said. Also, the threads _wake_ up.